fvm/src/machine/limiter.rs

```diff
@@ -1,137 +1,127 @@
 // Copyright 2021-2023 Protocol Labs
 // SPDX-License-Identifier: Apache-2.0, MIT
-use wasmtime::ResourceLimiter;
 
 use crate::machine::NetworkConfig;
 
 /// Execution level memory tracking and adjustment.
-pub trait ExecMemory
-where
-    Self: Sized,
-{
+pub trait MemoryLimiter: Sized {
     /// Get a snapshot of the total memory required by the modules on the call stack so far.
-    fn curr_exec_memory_bytes(&self) -> usize;
```

Comment:
I included `_bytes` so we know it's not pages for example. Ideally if doesn't say it's bytes, it should return some domain type like `struct Bytes(usize)`. If that's not desirable, at least the docstring could be updated to to say what the measure is.